src/Servers/Kestrel/Core/src/Internal/Http/Http1ContentLengthMessageBody.cs

```diff
@@ -90,14 +90,17 @@ public override async ValueTask<ReadResult> ReadAsyncInternal(CancellationToken
                     else
                     {
                         // Reset the timing read here for the next call to read.
-                        StopTimingRead(0);
+                        ReportReadBytes(0);
```

Comment:
I guess this is the one place where you don't need to call ReportReadBytes() when also calling StopTimingRead().

Reply:
Removed

Reply:
This is making me think short circuiting TimeoutControl.BytesRead when bytesRead is 0 might be a good idea whether or not we remove this particular call to ReportAndReadBytes. That way we could save a Monitor.Enter/Exit call. Not super important since I assume 0-byte request body reads a relatively rare, and the lock should not be very contested, though it could be contested by a timer tick in the background.

Reply:
Added short ciruit